### ISSUE TITLE
Any plan for windows os users?

### ISSUE THREAD

**Issue by MiyamuraMiyako** [wavetermdev/waveterm#76]
Any plan for windows os users?

**Comment by sawka** [comment]
Yes!  Windows support is definitely on the roadmap.  The plan is to support Windows via WSL, mainly as a way for windows users to log into and work on remote machines.  Think of it as very modern PuTTY replacement.

**Comment by earlev4** [comment]
> Yes! Windows support is definitely on the roadmap. The plan is to support Windows via WSL, mainly as a way for windows users to log into and work on remote machines. Think of it as very modern PuTTY replacement.

Hi @sawka! Thanks so much for the exciting project! That's wonderful news about support for WSL! You mention using the Windows version mainly as a way to "log into and work on remote machines". Do you also envision using it as a terminal to interact with the local WSL distribution (i.e. Ubuntu) installed on the Windows machine? Basically, could it serve as a replacement option for Windows Terminal application when using WSL?
Thanks again!

**Comment by celevra** [comment]
+1

**Comment by kwonoj** [comment]
I personally hope windows support includes host environment itself for the users using posix-like shell env via cygwin, or msys2.

**Comment by sawka** [comment]
> > Yes! Windows support is definitely on the roadmap. The plan is to support Windows via WSL, mainly as a way for windows users to log into and work on remote machines. Think of it as very modern PuTTY replacement.
> 
> Hi @sawka! Thanks so much for the exciting project! That's wonderful news about support for WSL! You mention using the Windows version mainly as a way to "log into and work on remote machines". Do you also envision using it as a terminal to interact with the local WSL distribution (i.e. Ubuntu) installed on the Windows machine? Basically, could it serve as a replacement option for Windows Terminal application when using WSL? Thanks again!

Yes, the plan is to allow interaction with the local WSL environment.

**Comment by sawka** [comment]
Okay, so we still plan on shipping a *native* Windows build in the future, but because we support ubuntu with our Linux build, it does look like you can get Wave working as a *WSL* app!  Some quick instructions on the [Quickstart](https://docs.waveterm.dev/quickstart) page.

If anyone on this issue thread tries it out, would love to get feedback on how your experience.

**Comment by MystPi** [comment]
I find it a bit strange that all of the new terminal projects recently (like Warp and Fig, and now Wave) do not support Windows natively, instead promising to support it in the future. Why is that? Windows is still the most popular OS in the world. Supporting it from the get-go would gain a lot more new users and give valuable feedback to the devs.

**Comment by Inplex-sys** [comment]
@MystPi Totally agree 🤝

**Comment by markox92** [comment]
There is one problem I think, windows already make "Windows Terminal" soo much better than any others you can find, and maybe is this problem for all developers, also there is Warp yes but who want terminal which need to login to use? I was try in on my Linux and after login popup i was uninstall warp.

**Comment by Jason-Linux** [comment]
+1

**Comment by Inplex-sys** [comment]
+1

**Comment by adamerose** [comment]
I’d love to use Wave as my main Windows terminal, as a replacement for Windows Terminal + Git Bash (MSYS).

I dislike WSL, it's a Linux VM with it's own file system and programs glued to the host on a best-effort basis which comes with a lot of bad UX. 

<details><summary>WSL Rant</summary>
I can't just copy paste file paths from File Explorer into the shell and have it work like it does in Git Bash, because everything is under /mnt and backslashes aren't supported. It uses separate installs of docker, python, virtual environments, environment variables, all needing to be installed and configured separately. If I run a docker container in Git Bash I can see it in Docker Desktop, in WSL it's two completely different docker installations. In WSL the git command refers to /usr/bin/git inside the VM which is invisible to Windows so tools GitKraken can't use it. If I create a Python venv for a project in WSL then try running it with my Windows Python installation it won't work. Performance is worse, notably with Git. Many one-liners/scripts don't work the same like `start chrome http://localhost`. Opening VS Code from WSL opens it as a remote tunnel using the WSL extension which has many nuances, like clicking Reveal in File Explorer in ~ takes me to \\wsl.localhost\Ubuntu\home\adam instead of C:\Users\Adam. I could go on... but anyways, I think the type of person bothered by these things is the exact same demographic that would be interested in using high-UX terminals like Wave, and turned away from using it if it's WSL only.
</details>

**Comment by staticfrost** [comment]
+1

**Comment by cogscides** [comment]
+1

**Comment by konosubakonoakua** [comment]
:rocket: + 1

**Comment by Krafman** [comment]
🚀 + 1

**Comment by Satishpokala124** [comment]
🚀 + 1

**Comment by no-1ankush** [comment]
> Okay, so we still plan on shipping a _native_ Windows build in the future, but because we support ubuntu with our Linux build, it does look like you can get Wave working as a _WSL_ app! Some quick instructions on the [Quickstart](https://docs.waveterm.dev/quickstart) page.
> 
> If anyone on this issue thread tries it out, would love to get feedback on how your experience.

Hi @sawka - Love Waveterm! Is there any further update on the _native_ Windows build?

**Comment by adam-the** [comment]
Hi @sawka - Is there any plans to support Powershell, or is Waveterm going to exclusively target WSL usage? Thanks

**Comment by flowstate247** [comment]
WSL is a bit overkill for me. What i'd love to see is a native Windows version of Wave that supports Git Bash or Cygwin.

**Comment by esimkowitz** [comment]
Updating this super late, but we have added native Windows, PowerShell, and WSL support in our recent releases

**Comment by RokeJulianLockhart** [comment]
> https://github.com/wavetermdev/waveterm/issues/76#issuecomment-2306626523

@adam-the, PowerShell works on Linux. I'm using it RN.
